### ISSUE TITLE
Automatic `aria-current` on `Link` and other components accepting `href` prop

### ISSUE THREAD

**Issue by arzatskis** [adobe/react-spectrum#5664]
### Provide a general summary of the feature here

Seems it's not there, event when following [client side routing guide](https://react-spectrum.adobe.com/react-aria/routing.html). Would be nice to have a easy, non-manual way to add this

### 🤔 Expected Behavior?

This could work same way as `react-router`s [`NavLink`](https://reactrouter.com/en/main/components/nav-link#aria-current) 

### 😯 Current Behavior

Currently you can only manually pass `<Link aria-current />` and `isCurrent` in render props will be available based on that, but you need to manually track that everywhere (maybe I'm missing something)

### 💁 Possible Solution

`aria-current` could be appended automatically or some kind of prop for `RouterProvider` to enable this maybe?

### 🔦 Context

This would be helpful not only for `Link`s in navigation, but also other elements accepting `href` prop. E.g. `<Row href="?preview=uuid" />` once clicked, panel is opened up with details and one could style current row based on that

**Comment by luisrudge** [comment]
@LFDanLu then it's impossible to replicate the `NavLink` behaviour from react-router, since `isCurrent` is always false when using Link+client side routing. Maybe RAC should have a NavLink as well, or help we figure out when the current link is active or not.

**Comment by snowystinger** [comment]
This should be solvable now with the `render` prop on `Link` and making use of the NavLink.
